kubetest/azure.go

```diff
@@ -0,0 +1,524 @@
+		return fmt.Errorf("failed to generate ARM templates: %v.", err)
+	}
+	return nil
+}
+
+func (c *Cluster) loadARMTemplates() error {
+	var err error
+	template, err := ioutil.ReadFile(path.Join(c.outputDir, "azuredeploy.json"))
+	if err != nil {
+		return fmt.Errorf("error reading ARM template file: %v.", err)
+	}
+	c.templateJSON = make(map[string]interface{})
+	err = json.Unmarshal(template, &c.templateJSON)
+	if err != nil {
+		return fmt.Errorf("error unmarshall template %v", err.Error())
+	}
+	parameters, err := ioutil.ReadFile(path.Join(c.outputDir, "azuredeploy.parameters.json"))
+	if err != nil {
+		return fmt.Errorf("error reading ARM parameters file: %v", err)
+	}
+	c.parametersJSON = make(map[string]interface{})
+	err = json.Unmarshal(parameters, &c.parametersJSON)
+	if err != nil {
+		return fmt.Errorf("error unmarshall parameters %v", err.Error())
+	}
+	c.parametersJSON = c.parametersJSON["parameters"].(map[string]interface{})
+
+	return nil
+}
+
+func (c *Cluster) getARMClient(ctx context.Context) error {
+	// instantiate Azure Resource Manager Client
+	env, err := azure.EnvironmentFromName("AzurePublicCloud")
+	var client *AzureClient
+	if client, err = getAzureClient(env,
+		c.credentials.SubscriptionID,
+		c.credentials.ClientID,
+		c.credentials.TenantID,
+		c.credentials.ClientSecret); err != nil {
+		return fmt.Errorf("error trying to get Azure Client: %v", err)
+	}
+	c.azureClient = client
+	return nil
+}
+
+func (c *Cluster) createCluster() error {
+	var err error
+	kubecfgDir, _ := ioutil.ReadDir(path.Join(c.outputDir, "kubeconfig"))
+	kubecfg := path.Join(c.outputDir, "kubeconfig", kubecfgDir[0].Name())
+	log.Printf("Setting kubeconfig env variable: kubeconfig path: %v.", kubecfg)
+	os.Setenv("KUBECONFIG", kubecfg)
+	log.Printf("Creating resurce group: %v.", c.resourceGroup)
+
+	log.Printf("Creating Azure resource group: %v for cluster deployment.", c.resourceGroup)
+	_, err = c.azureClient.EnsureResourceGroup(c.ctx, c.resourceGroup, c.location, nil)
+	if err != nil {
+		return fmt.Errorf("could not ensure resource group: %v", err)
+	}
+	log.Printf("Validating deployment ARM templates.")
+	if _, err := c.azureClient.ValidateDeployment(
+		c.ctx, c.resourceGroup, c.name, &c.templateJSON, &c.parametersJSON,
+	); err != nil {
+		return fmt.Errorf("ARM template invalid: %v", err)
+	}
+	log.Printf("Deploying cluster %v in resource group %v.", c.name, c.resourceGroup)
+	if _, err := c.azureClient.DeployTemplate(
+		c.ctx, c.resourceGroup, c.name, &c.templateJSON, &c.parametersJSON,
+	); err != nil {
+		return fmt.Errorf("cannot deploy: %v", err)
+	}
+	return nil
+
+}
+
+func (c *Cluster) buildHyperKube() error {
+
+	os.Setenv("VERSION", fmt.Sprintf("win-e2e-%v", os.Getenv("BUILD_NUMBER")))
+
+	cwd, _ := os.Getwd()
+	log.Printf("CWD %v", cwd)
+	push_hyperkube := util.K8s("kubernetes", "hack", "dev-push-hyperkube.sh")
```

Comment:
One thing I think might be an issue here is that dev-push-hyperkube.sh is using docker cli to push. But even with the service account preset, it might not be able to push into the specified gcr if a previous "docker auth" has not been performed. I haven't been able to figure out how the environment looks like from this perspective. @BenTheElder  what would your suggestion be here? I tried a combination of --build and --stage but it doesn't really build a hyperkube image as we need for acs-engine. A very hacky "gcloud auth configure-docker" right before calling the script ?

Reply:
`gcloud auth configure-docker` is something we were calling in the official kubernetes release scripts at one point, so that seems reasonable to me. Also sorry for the delay, I've been in crunch mode here with perf season, catching up on this and related PRs.